source/extensions/filters/http/peer_metadata/config.proto

```diff
@@ -71,4 +71,7 @@ message Config {
 
   // True to enable sharing with the upstream.
   bool shared_with_upstream = 5;
+
+  // Strip x-envoy-peer-metadata and x-envoy-peer-metadata-id headers on HTTP requests to services outside the mesh
+  bool strip_headers = 6;
```

Comment:
Can we move it to IstioHeaders message and call it "skip_external_clusters"? The comment should say it's detecting upstream clusters with "istio" "external" field.